frontend/packages/llama-stack-modular-ui/frontend/src/app/services/llamaStackService.ts

```diff
@@ -1,49 +1,114 @@
 /* eslint-disable camelcase */
 /* eslint-disable no-relative-import-paths/no-relative-import-paths */
-import type { Model as LlamaModel } from 'llama-stack-client/resources/models';
+import {
+  LlamaModel,
+  LlamaModelType,
+  VectorDB,
+  ChatbotSourceSettings,
+  Source,
+  Query,
+  QueryResponse,
+} from '../types';
 import axios from '../utilities/axios';
 
-// Roles must be 'user' and 'assistant' according to the Llama Stack API
-export type ChatMessage = {
-  role: 'user' | 'assistant';
-  content: string;
-  stop_reason?: string;
+/**
+ * Fetches all available models from the Llama Stack API
+ * @returns Promise<LlamaModel[]> - Array of available models with their metadata
+ * @throws Error - When the API request fails or returns an error response
+ */
+export const getModels = (): Promise<LlamaModel[]> => {
+  const url = '/api/v1/models';
+  return axios
+    .get(url)
+    .then((response) => response.data.data.items)
+    .catch((error) => {
+      throw new Error(error.response.data.message || 'Failed to fetch models');
+    });
 };
 
-export const listModels = (): Promise<LlamaModel[]> => {
-  const url = '/api/llama-stack/models/list';
+/**
+ * Fetches models filtered by a specific model type from the Llama Stack API
+ * @param modelType - The type of models to fetch (e.g., 'llm' or 'embedding')
+ * @returns Promise<LlamaModel[]> - Array of models matching the specified type
+ * @throws Error - When the API request fails or returns an error response
+ */
+export const getModelsByType = (modelType: LlamaModelType): Promise<LlamaModel[]> => {
+  const url = `/api/v1/models?model_type=${modelType}`;
```

Comment:
should we allow the case of an empty model type?
I think the backend method accepts an empty model type and returns all models iirc.

If not, we should filter it or check it properly.